lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -196,32 +196,30 @@ export default class BitcoinProcessor {
    */
   private async fastProcessTransactions (startingBlock: IBlockInfo) {
     const bitcoinBlockDataIterator = new BitcoinBlockDataIterator(this.bitcoinDataDirectory!);
-    const bestHeight = await this.bitcoinClient.getCurrentBlockHeight();
-    const bestHash = await this.bitcoinClient.getBlockHash(bestHeight);
-
-    let currentHash = bestHash;
-    let currentHeight = bestHeight;
-    const numOfBlocksToProcess = bestHeight - startingBlock.height + 1;
+    const lastBlockHeight = await this.bitcoinClient.getCurrentBlockHeight();
+    const lastBlockInfo = await this.bitcoinClient.getBlockInfoFromHeight(lastBlockHeight);
 
+    const numOfBlocksToProcess = lastBlockHeight - startingBlock.height + 1;
+    
     // An array of maps<hash, IBlockInfoExtended>.
     const chainInfos = [...new Array(numOfBlocksToProcess)].map<Map<string, IBlockInfoExtended>>(() => new Map());
-
-    console.log(`Begin fast processing block ${startingBlock.height} to ${bestHeight}`);
-    // loop through files and process them until we process all blocks needed
-    while (bitcoinBlockDataIterator.hasNext() && currentHeight >= startingBlock.height) {
-      const blockData = bitcoinBlockDataIterator.next()!;
-      await this.processBlockData(blockData, chainInfos, startingBlock.height, currentHeight);
-      [currentHeight, currentHash] = await this.markBestHeight(chainInfos, currentHeight, currentHash, startingBlock.height);
+    
+    console.log(`Begin fast processing block ${startingBlock.height} to ${lastBlockHeight}`);
+    // Loop through files backwards and process blocks from the end/tip of the blockchain until we reach the starting block given.
+    let hashOfOldestKnownValidBlock = lastBlockInfo.hash;
```

Comment:
earliest instead? Because we go from old to early and week keep updating it.

Reply:
Funny I started with earliest first. Will change it!